src/Spout/Writer/WriterFactory.php

```diff
@@ -20,7 +20,7 @@ class WriterFactory
      *
      * @api
      * @param  string $writerType Type of the writer to instantiate
-     * @return WriterInterface
+     * @return CSV\Writer | XLSX\Writer | ODS\Writer
```

Comment:
There are pros and cons to both methods. I actually had the enumeration of all possible types before but changed it to use the interface instead.

Pros of listing all types:
 - better autocomplete
 - more descriptive

Cons:
 - too much autocomplete (if you have a CSV writer, you should only get autocomplete for the CSV writer methods, not the XLSX ones)
 - harder to maintain (if you add a new type, you have to remember to add it here and in its reader counterpart)

I believe that the cons still outweigh the pros. I'd rather have less autocomplete but safer code (because all of the autocompleted methods are available). And when I need autocomplete, I add a PHPDoc hint on the variable itself. PHPStorm does support this very well.

For instance:
```php
/** @var \Box\Spout\...\CSV\Writer $writer */
$writer = WriterFactory::create(Type::CSV);

/** @var \Box\Spout\...\XLSX\Writer $writer */
$writer = WriterFactory::create(Type::XLSX);
```